src/sonic-yang-models/yang-models/sonic-portchannel.yang

```diff
@@ -74,8 +78,8 @@ module sonic-portchannel {
                 }
 
                 leaf min_links {
-                    type uint8 {
-                        range 1..128;
+                    type uint16 {
+                        range 0..1024;
                     }
```

Comment:
Shouldnt the range be 1..1024?

If we do support value 0, what does it indicate?

Reply:
Agree

Reply:
I found this necessary to be compatible with related https://github.com/Azure/sonic-utilities/pull/1630.  The original code there sets the default value of min-links to 0, and now that I have introduced range checking there, keeping a lower bound of 1 causes range check to fail.  So this change aligns yang with CLI.

Reply:
Did you test the 0 in current sonic image? Does teamd support it? ie. you set it as 0, and admin down all the member ports, and the portchannel itself is still up?

Reply:
Results are the same with current master and with the changes in this PR.  Recall that the default value of min-links in sonic-utilities is 0.  In both cases, shutting down all members of the LAG causes the portchannel to go oper down.  I noticed some non-deterministic behaviour, that may be a separate bug: with one of two member ports down, the LAG is sometimes up and sometimes down. 

Since the default is 0, one would expect that in the current master, a portchannel created with default min-links should be up when all members are down.  On the other hand, the work in this PR is supposed to clean up mismatches, and based on Qi's question, I now think that it doesn't do what it should.  What I did was align yang with CLI, but it seems teamd is now the odd one out, as the change in teamd only adjusted the upper limit and kept the lower at 1.  If you think it is reasonable to specify that a portchannel with all members down should be up (i.e. min-links = 0), then we have some more work to do to make teamd comply.  (Any other places?)  If not, then I should do the alignment the other way: put the lower limit in yang back to 1, and change the lower limit and default in sonic-utilities to 1.

Reply:
Thanks for the great insight! I prefer set lower limit of min-link to 1 to make life a little bit easier.

Reply:
I agree. Let's have the lower limit of min-links set to 1.